Tensile/KernelWriterAssembly.py

```diff
@@ -4932,44 +5223,46 @@ def lraFinalOffset(self, kernel, tP):
     #    hex(4-1), \
     #    vgpr(sgid), \
     #    "4 simds")
-
-    kStr += inst("v_mul_lo_u32", \
-        vgpr(sgid), \
-        sgpr(tmpSgpr), \
-        vgpr(sgid), \
-        "sgid=sgid*(MT%u+PAD)"%tIdx )
-
-    # TODO: generalize over different MIs
-    if kernel["MatrixInstruction"] and not kernel["ProblemType"]["DataType"].isHalf():
-      if tc == "B" and "MatrixInstK" in kernel and kernel["MatrixInstK"] > 1 and tP["bpe"] == 4:
-        kDiv = kernel["MatrixInstN"]
-        kStr += vectorStaticDivide(rReg, dividendReg, kDiv, tmpVgpr, tmpSgpr)
-        kStr += inst("v_and_b32", \
-          vgpr(rReg), \
-          # 2 = 64 / 32 or number of ks
-          hex(2-1), \
-          vgpr(rReg), \
-          "k groups")
-        kStr += inst("s_mov_b32", \
-          sgpr(tmpSgpr), \
-          hex(kernel["MacroTile%u"%tIdx] + kernel["LdsPad%s"%tc]), \
-          "MT%u+PAD"%tIdx )
-        kStr += inst("v_mul_lo_u32", \
-          vgpr(rReg), \
+ 
+    if not kernel["TransposeLDS"]:
+      kStr += inst("v_mul_lo_u32", \
+          vgpr(sgid), \
           sgpr(tmpSgpr), \
-          vgpr(rReg), \
-          "koff")
-        kStr += inst("v_add_u32", vgpr(sgid), vgpr(rReg), vgpr(sgid), "k offset")
-    # end TODO
-    if kernel["VectorWidth"] > 1 and not kernel["MatrixInstruction"]:
-      kStr += staticMultiply(vgpr(tP["gpr"]["lro"]), vgpr(tP["gpr"]["lro"]), \
-          kernel["VectorWidth"], sgpr(tmpSgpr))
-    kStr += inst("_v_add_lshl_u32", \
-        vgpr("LocalReadAddr%s"%tc), \
-        vgpr(sgid), \
-        vgpr(tP["gpr"]["lro"]), \
-        hex(log2(tP["bpe"])), \
-        "o = (lro%s*VW+sgid*MT%u)*bpe"%(tc, tIdx) )
+          vgpr(sgid), \
+          "sgid=sgid*(MT%u+PAD)"%tIdx )
+
+      # TODO: generalize over different MIs
+      if kernel["MatrixInstruction"] and not kernel["ProblemType"]["DataType"].isHalf():
+        if tc == "B" and "MatrixInstK" in kernel and kernel["MatrixInstK"] > 1 and tP["bpe"] == 4:
+          kDiv = kernel["MatrixInstN"]
+          kStr += vectorStaticDivide(rReg, dividendReg, kDiv, tmpVgpr, tmpSgpr)
+          kStr += inst("v_and_b32", \
+              vgpr(rReg), \
+              # 2 = 64 / 32 or number of ks
+              hex(2-1), \
```

Comment:
Always 2-1? k can be 4 or 16 for smaller instructions

Reply:
for f32 , that code will break in 16x16.. not sure who coded that.. definitely requires change..